### PR TITLE
elk7: 7.5.1 -> 7.10.2

### PR DESCRIPTION
**Pull request by happysalada**
###### Motivation for this change

upstream change.

There is one weird thing with logstash, the version is not available for some reason. I've opened an issue upstream.
https://github.com/elastic/logstash/issues/13176

I still feel we should upgrade, and we can just fix logstash when upstream solves the issue.

the nixpkgs-fmt commits are optional. If you feel strongly against them, I'm happy to remove them.

I haven't updated the linux sha256 yet.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 136137` at b87ec9d9 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/5d78b4daf624a1f069716d1859149c40">elasticsearch7</a></li>
    <li><a href="https://gist.github.com/250b27ddbb81e3bcc9c93cdebf4f35cf">filebeat7</a></li>
    <li><a href="https://gist.github.com/493dc55832aba888cce07f3134153f82">heartbeat7</a></li>
    <li><a href="https://gist.github.com/565fced2278a8cf8f5d4bfdf3e05f70b">journalbeat7</a></li>
    <li><a href="https://gist.github.com/49d9758aaaa4ee32153f1787337ba32a">kibana7</a></li>
    <li><a href="https://gist.github.com/d6ae47fb4a861c8e1b0c50c75c18b710">metricbeat7</a></li>
    <li><a href="https://gist.github.com/2a82a28d47f2763e36b40276998ea958">packetbeat7</a></li>
  </ul>
</details>
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>elasticsearch7-oss</li>
    <li>kibana7-oss</li>
    <li>logstash7</li>
    <li>logstash7-oss</li>
  </ul>
</details>
<details>
  <summary>10 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/servers/search/elasticsearch/7.x.nix:40:15:
```
   |
40 |   patches = [ ./es-home-6.x.patch ];
   |               ^
```
</li>
    <li>warning: unused-argument

Unused argument: `elk7Version`.
Near pkgs/tools/misc/logstash/7.x.nix:1:3:
```
  |
1 | { elk7Version
  |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/servers/search/elasticsearch/7.x.nix:56:3:
```
   |
56 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

makeWrapper is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/tools/misc/logstash/7.x.nix:31:3:
```
   |
31 |   buildInputs = [
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md">name-and-version</a>

Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/tools/misc/logstash/7.x.nix:16:3:
```
   |
16 |   name = "logstash-${optionalString (!enableUnfree) "oss-"}${version}";
   |   ^
```

Near pkgs/tools/misc/logstash/7.x.nix:15:3:
```
   |
15 |   version = "7.9.3";
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/go-packages/generic/default.nix:257:5:
```
    |
257 |     meta = {
    |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md">name-and-version</a>

Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/misc/logging/beats/7.x.nix:4:3:
```
  |
4 |   name = "${package}-${version}";
  |   ^
```

Near pkgs/misc/logging/beats/7.x.nix:5:3:
```
  |
5 |   version = elk7Version;
  |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md">license-missing</a>

Package is missing a license.

Near pkgs/development/go-packages/generic/default.nix:257:5:
```
    |
257 |     meta = {
    |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/misc/kibana/7.x.nix:49:3:
```
   |
49 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md">name-and-version</a>

Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/development/tools/misc/kibana/7.x.nix:32:3:
```
   |
32 |   name = "kibana-${optionalString (!enableUnfree) "oss-"}${version}";
   |   ^
```

Near pkgs/development/tools/misc/kibana/7.x.nix:33:3:
```
   |
33 |   version = elk7Version;
   |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/tools/misc/kibana/7.x.nix [hunk 1]
```diff
@@ -18,15 +19,16 @@ let
   shas =
     if enableUnfree
     then {
-      x86_64-linux  = "1wq4fc2fifkg1qz7nxdfb4yi2biay8cgdz7kl5k0p37sxn0sbkja";
-      x86_64-darwin = "06346kj7bv49py49pmmnmh8m24322m88v1af19909pj9cxgd0p6v";
+      x86_64-linux = "1wq4fc2fifkg1qz7nxdfb4yi2biay8cgdz7kl5k0p37sxn0sbkja";
```
Comment (fufexan) [inline pkgs/development/tools/misc/kibana/7.x.nix:22]:
```suggestion
      x86_64-linux = "sha256-lTPBppKm51zgKSQtSdO0PgZ/aomvaStwqwYYGNPY4Bo=";
```

File: pkgs/servers/search/elasticsearch/7.x.nix [hunk 2]
```diff
@@ -17,12 +20,12 @@ let
   shas =
     if enableUnfree
     then {
-      x86_64-linux  = "1s27bzx5y8vcd95qrw6av3fhyxb45219x9ahwaxa2cygmbpighrp";
-      x86_64-darwin = "1ia3byir3i5qaarmcaysrg3dhnxjmxnf0m0kzyf61g9aiy87gb7q";
+      x86_64-linux = "1s27bzx5y8vcd95qrw6av3fhyxb45219x9ahwaxa2cygmbpighrp";
```
Comment (fufexan) [inline pkgs/servers/search/elasticsearch/7.x.nix:23]:
```suggestion
      x86_64-linux = "sha256-O3rjtvXyJI+kRBqiz2U2OMkCIQj4E+AIHaE8N4o14R4=";
```

**Review by fufexan — COMMENTED** [review]
It seems like all the beats have a similar output when building. - https://gist.github.com/fufexan/6ee89afeca6aa5bc9ee1c9a2957b37ba

**Comment by happysalada** [comment]
@fufexan thanks a lot for the help!
I don't think the beats need a hash update. It seems to be the same on darwin and linux.
I'll squash the commits when all the hashes have been added.
I'll try to retain authorship. I think leaving the authored-by in the commit message should do that.

**Comment by happysalada** [comment]
@fufexan could I ask for a little more help? There are also the hash of `elasticsearch7-oss` and `kibana7-oss` that need an update.

**Comment by fufexan** [comment]
@happysalada I messaged you on Matrix on SoN General, and explained that somehow building both those packages goes fine, with no hash mismatches. It's weird, but I don't know how it's happening :sweat_smile:

**Comment by happysalada** [comment]
Oh, sorry I don't check matrix when I'm trying to code.
You probably have to modify the hashes. replace those with empty version `""` or just modify one letter and it should try to rebuild those.
When a hash is not changed, even if the version hash changed sometimes nix will just use the version from the binary cache.
I'm going to connect on matrix in case there is anything else that is weird.

File: pkgs/development/tools/misc/kibana/7.x.nix [hunk 3]
```diff
@@ -18,15 +19,16 @@ let
...
     then {
-      x86_64-linux  = "1wq4fc2fifkg1qz7nxdfb4yi2biay8cgdz7kl5k0p37sxn0sbkja";
-      x86_64-darwin = "06346kj7bv49py49pmmnmh8m24322m88v1af19909pj9cxgd0p6v";
+      x86_64-linux = "sha256-lTPBppKm51zgKSQtSdO0PgZ/aomvaStwqwYYGNPY4Bo=";
+      x86_64-darwin = "sha256-d7xHmoASiywDlZCJX/CfUX1VIi4iOcDrqvK0su54MJc=";
     }
     else {
-      x86_64-linux  = "0ygpmcm6wdcnvw8azwqc5257lyic7yw31rqvm2pw3afhpha62lpj";
-      x86_64-darwin = "0xy81g0bhxp47p29kkkh5llfzqkzqzr5dk50ap2hy0hjw33ld6g1";
+      x86_64-linux = "0ygpmcm6wdcnvw8azwqc5257lyic7yw31rqvm2pw3afhpha62lpj";
```
Comment (fufexan) [inline pkgs/development/tools/misc/kibana/7.x.nix:26]:
```suggestion
      x86_64-linux = "sha256-+pkKpiXBpLHs72KKNtMJbqipw6eu5XC1xu/iLFCHGRQ=";
```

File: pkgs/servers/search/elasticsearch/7.x.nix [hunk 4]
```diff
@@ -17,12 +20,12 @@ let
...
     then {
-      x86_64-linux  = "1s27bzx5y8vcd95qrw6av3fhyxb45219x9ahwaxa2cygmbpighrp";
-      x86_64-darwin = "1ia3byir3i5qaarmcaysrg3dhnxjmxnf0m0kzyf61g9aiy87gb7q";
+      x86_64-linux = "sha256-O3rjtvXyJI+kRBqiz2U2OMkCIQj4E+AIHaE8N4o14R4=";
+      x86_64-darwin = "sha256-AwuY2yMxf+v7U5/KD3Cf+Hv6ijjySEyj6pzF3RCsg24=";
     }
     else {
-      x86_64-linux  = "005i7d7ag10qkn7bkx7md50iihvcvc84hay2j94wvsm7yghhbmi3";
-      x86_64-darwin = "01f81720rbzdqc0g1xymhz2lflldfbnb0rh7mpki99pss28vj9sh";
+      x86_64-linux = "005i7d7ag10qkn7bkx7md50iihvcvc84hay2j94wvsm7yghhbmi3";
```
Comment (fufexan) [inline pkgs/servers/search/elasticsearch/7.x.nix:27]:
```suggestion
      x86_64-linux = "sha256-cJrdkFIFgAI6wfQh34Z8yFuLrOCOKzgOsWZhU3S/3NQ=";
```

**Comment by happysalada** [comment]
Just upgraded a server from the old version to the new version for elastic search and kibana and can confirm the upgrade went without a hiccup.
I'm leaving this a little more open in case somebody is interested.

**Comment by happysalada** [comment]
@fufexan upstream responded regarding the logstash url change.
Would you mind trying to build logstash for linux (oss and non oss version) and commenting with the hashes ?
this should be the last time.

File: pkgs/tools/misc/logstash/7.x.nix [hunk 5]
```diff
@@ -9,25 +10,38 @@
...
 
-let this = stdenv.mkDerivation rec {
+let
+  info = splitString "-" stdenv.hostPlatform.system;
+  arch = elemAt info 0;
+  plat = elemAt info 1;
+  shas =
+    if enableUnfree
+    then {
+      x86_64-linux = "sha256-lTPBppKm51zgKSQtSdO0PgZ/aomvaStwqwYYGNPY4Bo=";
```
Comment (fufexan) [inline pkgs/tools/misc/logstash/7.x.nix:20]:
```suggestion
      x86_64-linux = "sha256-5qv4fbFpLf6aduD7wyxXQ6FsCeUqrszRisNBx44vbMY=";
```

File: pkgs/tools/misc/logstash/7.x.nix [hunk 6]
```diff
@@ -9,25 +10,38 @@
...
+      x86_64-darwin = "sha256-7H+Xpo8qF1ZZMkR5n92PVplEN4JsBEYar91zHQhE+Lo=";
+    }
+    else {
+      x86_64-linux = "sha256-+pkKpiXBpLHs72KKNtMJbqipw6eu5XC1xu/iLFCHGRQ=";
```
Comment (fufexan) [inline pkgs/tools/misc/logstash/7.x.nix:24]:
```suggestion
      x86_64-linux = "sha256-jiV2yGPwPgZ5plo3ftImVDLSOsk/XBzFkeeALSObLhU=";
```

**Comment by happysalada** [comment]
There seems to be no regression as far as I've tested, I'm happy to do follow up PRs if anybody finds errors.

**Pull request merged**